### ISSUE TITLE
Would be nice if all models were json serializable

### ISSUE THREAD

**Issue by sgehrman** [amugofjava/podcast_search#17]
I want to pass from backend to frontend, and it would be easier if they would serialize to json.

**Comment by amugofjava** [comment]
Hi @sgehrman,

This is an interesting idea. Are you thinking of just being able to serialise the models to JSON, or also thinking of de-serialising JSON back into the models?

**Comment by loveoverflowdev** [comment]
Flutter doesn't support Reflection like Java. So you cannot see any framework like GSON.
Actually, Dart language supports mirror, which is similar to Reflection, but this feature was ignored by Flutter

**Comment by amugofjava** [comment]
You're right, Flutter doesn't support reflection but there are ways this request can be achieved. You can serialise/deserialise objects to/from JSON manually (my preferred approach) or via code gen using a package such as [json_serializable](https://pub.dev/packages/json_serializable).
